build-combo.sh

```diff
@@ -0,0 +1,104 @@
+#! /bin/bash
+set -eo pipefail
+trap ctrl_c INT
+
+readonly YELLOW=$(tput setaf 3)
+readonly NC=$(tput sgr0)
+
+readonly BACKUP_FILE=".dazzle.yaml.orig"
+readonly ORIGINAL_FILE="dazzle.yaml"
+
+function save_original {
+    if [ ! -f ${BACKUP_FILE} ]; then
+        echo "${YELLOW}Creating a backup of ${ORIGINAL_FILE} as it does not exist yet...${NC}" && cp ${ORIGINAL_FILE} ${BACKUP_FILE}
+    fi
+}
+
+function restore_original() {
+	echo "${YELLOW}Restoring backup file ${BACKUP_FILE} to original file ${ORIGINAL_FILE}${NC}"
+	cp ${BACKUP_FILE} ${ORIGINAL_FILE}
+}
+
+function ctrl_c() {
+	echo "${YELLOW}** Trapped CTRL-C${NC}"
+	restore_original
+}
+
+function build_combination {
+    combination=$1
+
+    local exists="$(yq e '.combiner.combinations[] | select (.name=="'"$combination"'")' dazzle.yaml)"
+    if [[ -z "$exists" ]]; then
+        echo "combination is not defined"
+        exit 1
+    fi
+
+    refs=$(get_refs "$combination")
+    requiredChunks=$(get_chunks "$refs" | sort | uniq)   
+    availableChunks=$(get_available_chunks)
+
+    for ac in $availableChunks; do
+        if [[ ! "${requiredChunks[*]}" =~ "${ac}" ]]; then
+            dazzle project ignore "$ac"
+        fi
+    done
+}
+
+function get_refs {
+    local ref=$1
+    echo "$ref"
+
+    refs="$(yq e '.combiner.combinations[] | select (.name=="'"$ref"'") | .ref[]' dazzle.yaml)"
+    if [[ -z "$refs" ]]; then
+        return
+    fi
+
+    for ref in $refs; do
+        get_refs "$ref"
+    done
+}
+
+function get_chunks {
+    for ref in $@; do
```

Comment:
```suggestion
    for ref in "$@"; do
```

based on vs code recommendation

Reply:
Had tried that but the loop is not executed if I change it.